Add section about syntax special characters

Hunterness opened the pull request:
We're using a custom syntax for describing the administration commands but we never explain what our special characters mean, for example `{ ... }` or `[ ... ]`.

Hunterness committed:
Update relationship constraint syntax

So that the `[` in the pattern is not confused with the `[` indicating optional

Hunterness committed:
Add section about syntax special characters

Lojjs commented on cypher/cypher-docs/src/docs/dev/ql/administration/security/index.asciidoc:
```diff
@@ -86,6 +87,69 @@ Instead they are defined in terms of their underlying _privileges_ and they can
+
+The special characters and their meaning are as follows:
+
+.Special characters in syntax summeries
```
Very nice table, I think it will be really helpful for understanding the syntax boxes.

Lojjs commented on cypher/cypher-docs/src/docs/dev/ql/administration/security/index.asciidoc:
```diff
@@ -86,6 +87,69 @@ Instead they are defined in terms of their underlying _privileges_ and they can
+Some things to notice about this command is that it includes `{` and `}` in the syntax, and between them has a grouping of either a list of properties or the character `*`.
+It also has multiple optional parts, including the entity part of the command which is the grouping following the graph name.
+
+In difference, there is no need to escape any characters in the node property existence constraint creation command.
```
Not sure how, but we should maybe clarify here that we mean that there is no need to escape it in the syntax boxes in the documentation. Normally we refer to the backticks in Cypher when we talk about escaping, and this is something quite different.

Hunterness replied:
True, not sure how either though

Lojjs commented on cypher/cypher-docs/src/docs/dev/ql/administration/security/index.asciidoc:
```diff
@@ -86,6 +87,69 @@ Instead they are defined in terms of their underlying _privileges_ and they can
+| A comma separated list of names would be `name[, ...]`.
+
+| `"`
+| When a special character is part of the syntax itself, we surround it with `"` to indicate this.
```
In general I agree that we must come up with a way to distinguish between special characters that have a specific meaning for syntax like optional and special characters that is actually part of the command. I'm not sure if this is the best way to do it though. At least I would like `\` better as an escape character than `"`. Maybe someone in docs team have an idea?

Hunterness replied:
I think we got the `"` from PRDs, and it is used in 4.1 for the properties in the [READ privilege](https://neo4j.com/docs/cypher-manual/current/administration/security/reads/#administration-security-reads-read) which is why I have that here

I just documented what we were using :P Though I agree that it isn't the nicest

Hunterness replied:
If we decide to change it to something else we need to update both where it already exists and some of my currently open PRs :P

Lojjs reviewed (commented):
Overall I think this kind of explaining text and table is super helpful to understand the syntax boxes. But I'm not super fond of the way of escaping characters with `"` and I'm also unsure if all the text is super clear for someone how is unfamiliar with this. Since that is more a matter of language and pedagogics, I leave it to someone in docs team to judge.

Hunterness commented on cypher/cypher-docs/src/docs/dev/ql/administration/security/index.asciidoc:
```diff
@@ -86,6 +87,69 @@ Instead they are defined in terms of their underlying _privileges_ and they can
+
+| `\|`
+| Or, used to indicate alternative parts of a command. Needs to be part of a grouping.
+| If the syntax needs to specify either a name or `*`, this can be indicated with `name \| *`.
```
```suggestion
| If the syntax needs to specify either a name or `*`, this can be indicated with `* \| name`.
```
To make the order more consistent (have a card to update other places)

Hunterness commented on cypher/cypher-docs/src/docs/dev/ql/administration/security/index.asciidoc:
```diff
@@ -86,6 +87,69 @@ Instead they are defined in terms of their underlying _privileges_ and they can
+
+| `{` and `}`
+| Used to group parts of the command, common together with `\|`.
+| To use the `or` in the syntax summary it needs to be in a group, `{name \| *}`.
```
```suggestion
| To use the `or` in the syntax summary it needs to be in a group, `{* \| name}`.
```

Hunterness commented on cypher/cypher-docs/src/docs/dev/ql/administration/security/index.asciidoc:
```diff
@@ -86,6 +87,69 @@ Instead they are defined in terms of their underlying _privileges_ and they can
+
+| `"`
+| When a special character is part of the syntax itself, we surround it with `"` to indicate this.
+| To include `{` in the syntax use `"{" { name \| * } "}"`, here we get either `{*}` or `{name}`.
```
```suggestion
| To include `{` in the syntax use `"{" { * \| name } "}"`, here we get either `{*}` or `{name}`.
```

Hunterness commented on cypher/cypher-docs/src/docs/dev/ql/administration/security/index.asciidoc:
```diff
@@ -86,6 +87,69 @@ Instead they are defined in terms of their underlying _privileges_ and they can
+----
+GRANT READ
+    "{" { * | property[, ...] } "}"
+    ON {DEFAULT GRAPH | GRAPH[S] {name[, ...] | *}}
```
```suggestion
    ON {DEFAULT GRAPH | GRAPH[S] {* | name[, ...]}}
```

Hunterness committed:
Change order of name and *

To make the order more consistent in the Administration chapter (have a card to update other places)

Hunterness commented on cypher/cypher-docs/src/docs/dev/ql/administration/security/index.asciidoc:
```diff
@@ -86,6 +87,69 @@ Instead they are defined in terms of their underlying _privileges_ and they can
+----
+GRANT READ
+    "{" { * | property[, ...] } "}"
+    ON {DEFAULT GRAPH | GRAPH[S] {* | name[, ...]}}
```
```suggestion
    ON {DEFAULT GRAPH | GRAPH[S] { * | name[, ...] }}
```

Hunterness committed:
Add spaces for consistency

Hunterness committed:
Clean-up names

to be consistent with (cherry-picks of) https://github.com/neo4j/neo4j-documentation/pull/825

Hunterness commented:
After the rebase I got one failing tests that looks like it's [failing on general 4.3](https://live.neo4j-build.io/project.html?tab=testDetails&testNameId=-5717109592674643058&projectId=Documentation_Neo4jDocumentationTests&buildTypeId=Documentation_Neo4jDocumentationTests_Neo4j43DocsTest) as well

davidoliverSP2 committed:
editorial update
